feat(sbb-menu): introduce sbb-menu component (#1378)
lyne-design-system/lyne-components: src/components/sbb-menu/readme.md

> The `sbb-menu` is a component that you can attach to any element (trigger) to display a context menu. The menu appears on trigger left click. On mobile, the menu is displayed as a sheet with a backdrop; on desktop it will be shown as a floating menu and will calucate the optimal position relative to the trigger element by evaluating the available space with the following priority: start/below, start/above, end/below, end/above.

Reviewer: Maybe split in more lines?

# sbb-menu

> The `sbb-menu` is a component that you can attach to any element (trigger) to display a context menu. 

Reviewer: ```suggestion
The `sbb-menu` is a component that can be attached to any element (trigger) to display a context menu. 
```

The menu appears on trigger left click. 

On mobile, the menu is displayed as a sheet with a backdrop; 
on desktop it will be shown as a floating menu and will calculate the optimal position relative to the trigger element 
by evaluating the available space with the following priority: start/below, start/above, end/below, end/above.

## Usage

The menu component allows you to present a custom menu that allows you to perform actions relevant to the current task 
or to navigate within or outside the application by using the `sbb-menu-action` component along with it 

> or providing custom content inside the dialog as shown below:

Reviewer: ```suggestion
or providing custom content inside the component as shown below:
```

```html
<!-- Trigger element -->
<sbb-button id="menu-trigger" label="Menu trigger"></sbb-button>

<!-- Menu component with custom content and menu actions -->
<sbb-menu trigger="menu-trigger">
    <div>Christina Müller</div>
    <span>UIS9057</span>
    <sbb-link href="https://www.sbb.ch/en" negative text-size="xs" variant="block">Profile</sbb-link>
    
    <sbb-divider />

    <sbb-menu-action icon="link-small" href="https://www.sbb.ch/en">View</sbb-menu-action>
    <sbb-menu-action icon="pen-small">Edit</sbb-menu-action>
    <sbb-menu-action icon="swisspass-small" amount="123">Details</sbb-menu-action>
    
    <sbb-divider />
    
    <sbb-menu-action icon="cross-small">Cancel</sbb-menu-action>
</sbb-menu>
```

Clicking in the backdrop or pressing the `ESC` key closes the menu.

### Accessibility

> As the menu opens, the focus will automatically be set to the first focusable item within the dialog.

Reviewer: ```suggestion
As the menu opens, the focus will automatically be set to the first focusable item within the component.
```
